src/main/java/org/springframework/retry/support/RetryTemplateBuilder.java

```diff
@@ -132,6 +136,36 @@ public RetryTemplateBuilder withinMillis(long timeout) {
 		return this;
 	}
 
+	/**
+	 * Retry until {@code timeout} has passed since the initial attempt.
+	 * @param timeoutMillis timeout in milliseconds
+	 * @return this
+	 * @see TimeoutRetryPolicy
+	 * @throws IllegalArgumentException if timeout is {@literal <=} 0 or if another retry
+	 * policy has already been configured.
+	 * @since 2.0.2
+	 */
+	public RetryTemplateBuilder withTimeout(long timeoutMillis) {
+		Assert.isTrue(timeoutMillis > 0, "timeoutMillis should be greater than 0");
+		Assert.isNull(this.baseRetryPolicy, "You have already selected another retry policy");
+		this.baseRetryPolicy = new TimeoutRetryPolicy(timeoutMillis);
+		return this;
+	}
+
+	/**
+	 * Retry until {@code timeout} has passed since the initial attempt.
+	 * @param timeout duration for how long retries should be attempted
+	 * @return this
+	 * @see TimeoutRetryPolicy
+	 * @throws IllegalArgumentException if timeout is {@code null} or 0, or if another
+	 * retry policy has already been configured.
+	 * @since 2.0.2
+	 */
+	public RetryTemplateBuilder withTimeout(Duration timeout) {
+		Assert.notNull(timeout, "timeout is null");
+		return this.withinMillis(timeout.toMillis());
```

Comment:
No `this.` on method calls, please.
And why do you call a deprecated method from here then?